diff -u arch/um/drivers.old/ethertap_user.c arch/um/drivers/ethertap_user.c
--- arch/um/os-Linux/drivers.old/ethertap_user.c	Mon Apr 29 20:59:30 2002
+++ arch/um/os-Linux/drivers/ethertap_user.c	Mon Apr 29 21:00:27 2002
@@ -93,9 +93,9 @@
 	char version_buf[sizeof("nnnnn\0")];
 	char data_fd_buf[sizeof("nnnnnn\0")];
 	char gate_buf[sizeof("nnn.nnn.nnn.nnn\0")];
-	char *setup_args[] = { "uml_net", version_buf, "ethertap", dev,
+	char *setup_args[] = { "/usr/lib/uml/uml_net", version_buf, "ethertap", dev,
 			       data_fd_buf, gate_buf, NULL };
-	char *nosetup_args[] = { "uml_net", version_buf, "ethertap", 
+	char *nosetup_args[] = { "/usr/lib/uml/uml_net", version_buf, "ethertap", 
 				 dev, data_fd_buf, NULL };
 	char **args, c;
 
diff -u arch/um/drivers.old/net_user.c arch/um/drivers/net_user.c
--- arch/um/drivers.old/net_user.c	Mon Apr 29 20:59:30 2002
+++ arch/um/drivers/net_user.c	Mon Apr 29 21:00:59 2002
@@ -185,7 +185,7 @@
 	char addr_buf[sizeof("255.255.255.255\0")];
 	char netmask_buf[sizeof("255.255.255.255\0")];
 	char version[sizeof("nnnnn\0")];
-	char *argv[] = { "uml_net", version, what, dev, addr_buf, 
+	char *argv[] = { "/usr/lib/uml/uml_net", version, what, dev, addr_buf, 
 			 netmask_buf, NULL };
 	char *output;
 	int output_len, pid;
diff -u arch/um/drivers.old/slip_user.c arch/um/drivers/slip_user.c
--- arch/um/drivers.old/slip_user.c	Mon Apr 29 20:59:30 2002
+++ arch/um/drivers/slip_user.c	Mon Apr 29 21:01:12 2002
@@ -113,7 +113,7 @@
 	struct slip_data *pri = data;
 	char version_buf[sizeof("nnnnn\0")];
 	char gate_buf[sizeof("nnn.nnn.nnn.nnn\0")];
-	char *argv[] = { "uml_net", version_buf, "slip", "up", gate_buf, 
+	char *argv[] = { "/usr/lib/uml/uml_net", version_buf, "slip", "up", gate_buf, 
 			 NULL };
 	int sfd, mfd, disc, sencap, err;
 
@@ -166,7 +166,7 @@
 {
 	struct slip_data *pri = data;
 	char version_buf[sizeof("nnnnn\0")];
-	char *argv[] = { "uml_net", version_buf, "slip", "down", pri->name, 
+	char *argv[] = { "/usr/lib/uml/uml_net", version_buf, "slip", "down", pri->name, 
 			 NULL };
 	int err;
 
diff -u arch/um/drivers.old/tuntap_user.c arch/um/drivers/tuntap_user.c
--- arch/um/os-Linux/drivers.old/tuntap_user.c	Mon Apr 29 20:59:30 2002
+++ arch/um/os-Linux/drivers/tuntap_user.c	Mon Apr 29 21:01:22 2002
@@ -68,7 +68,7 @@
 {
 	struct tuntap_pre_exec_data data;
 	char version_buf[sizeof("nnnnn\0")];
-	char *argv[] = { "uml_net", version_buf, "tuntap", "up", gate,
+	char *argv[] = { "/usr/lib/uml/uml_net", version_buf, "tuntap", "up", gate,
 			 NULL };
 	char buf[CMSG_SPACE(sizeof(*fd_out))];
 	struct msghdr msg;
