diff --git a/arch/x86/um/Makefile b/arch/x86/um/Makefile
index 8fb5840..f322503 100644
--- a/arch/x86/um/Makefile
+++ b/arch/x86/um/Makefile
@@ -19,7 +19,8 @@ ifeq ($(CONFIG_X86_32),y)
 obj-y += checksum_32.o
 obj-$(CONFIG_BINFMT_ELF) += elfcore.o
 
-subarch-y = ../lib/string_32.o ../lib/atomic64_32.o ../lib/atomic64_cx8_32.o
+subarch-y = ../lib/string_32.o ../lib/atomic64_32.o ../lib/atomic64_cx8_32.o \
+	../lib/atomic64_386_32.o ../lib/cmpxchg8b_emu.o
 subarch-$(CONFIG_RWSEM_XCHGADD_ALGORITHM) += ../lib/rwsem.o
 subarch-$(CONFIG_HIGHMEM) += ../mm/highmem_32.o

